### PR TITLE
Added functions to complete the set of local movement translations

### PR DESCRIPTION
**Pull request by zacnomore**
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/537)

### CODE REVIEW

File: amethyst_core/src/transform/components/local_transform.rs [hunk 1]
```diff
@@ -64,16 +59,36 @@ impl LocalTransform {
...
+    }
+
+    /// Move up relative to current position and orientation.
     pub fn move_up(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
         self.move_local(orientation.up.into(), amount)
     }
 
+    /// Move down relative to current position and orientation.
+    pub fn move_down(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
+        self.move_up( orientation, -amount)
```
Comment (omni-viral) [inline amethyst_core/src/transform/components/local_transform.rs]: Why does down is implemented via up while other pairs are implemented independently?

**Comment by zacnomore** [comment]
I'm not sure why I did that. Changes completed

File: amethyst_core/src/transform/components/local_transform.rs [hunk 2]
```diff
@@ -64,16 +59,36 @@ impl LocalTransform {
...
+    }
+
+    /// Move up relative to current position and orientation.
     pub fn move_up(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
         self.move_local(orientation.up.into(), amount)
     }
 
+    /// Move down relative to current position and orientation.
+    pub fn move_down(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
+         self.move_local((-orientation.up).into(), amount)
```
Comment (omni-viral) [inline amethyst_core/src/transform/components/local_transform.rs]: Whitespace

Comment (zacnomore) [reply amethyst_core/src/transform/components/local_transform.rs]: Thank you. Did that add the change or do I need to do it myself?

Comment (omni-viral) [reply amethyst_core/src/transform/components/local_transform.rs]: You need to do it yourself

**Comment by zacnomore** [comment]
Not sure what to do, the whitespace seems fixed to me but I've made no change. Am I looking in the wrong [place](https://github.com/amethyst/amethyst/pull/537/files)?

```
    /// Move forward relative to current position and orientation.
    pub fn move_forward(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
        self.move_local(orientation.forward.into(), amount)
    }

    /// Move backward relative to current position and orientation.
    pub fn move_backward(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
        self.move_local((-orientation.forward).into(), amount)
    }

    /// Move right relative to current position and orientation.
    pub fn move_right(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
        self.move_local(orientation.right.into(), amount)
    }

    /// Move left relative to current position and orientation.
    pub fn move_left(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
        self.move_local((-orientation.right).into(), amount)
    }

    /// Move up relative to current position and orientation.
    pub fn move_up(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
        self.move_local(orientation.up.into(), amount)
    }

    /// Move down relative to current position and orientation.
    pub fn move_down(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
         self.move_local((-orientation.up).into(), amount)
    }
```

**Comment by omni-viral** [comment]
@zacnomore you see how `self` is aligned to `fn` except for  `move_down`?

**Comment by zacnomore** [comment]
OHHHHHHHHHH

**Comment by Rhuagh** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/amethyst/amethyst/537)*

**Comment by zacnomore** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[amethyst_core/src/transform/components/local_transform.rs, line 89 at r1](https://reviewable.io:443/reviews/amethyst/amethyst/537#-L2ekMvNiH2ISvnAQ3Dq-r1-89:-L2ul2Pu_bAPWvRtm74Q:b-896fix) ([raw file](https://github.com/amethyst/amethyst/blob/2132382fe4d717e7a3b9cdb303cd20cfde5f67c0/amethyst_core/src/transform/components/local_transform.rs#L89)):*
<details><summary><i>Previously, omni-viral (Zakarum) wrote…</i></summary><blockquote>

Why does down is implemented via up while other pairs are implemented independently?
</blockquote></details>

Done.

---

*[amethyst_core/src/transform/components/local_transform.rs, line 89 at r2](https://reviewable.io:443/reviews/amethyst/amethyst/537#-L2ekMvNiH2ISvnAQ3Dq-r2-89:-L2ul2rjoomOW0a1jzwo:b-896fix) ([raw file](https://github.com/amethyst/amethyst/blob/1c66d53fee74238a1519f60b57e898470df58d08/amethyst_core/src/transform/components/local_transform.rs#L89)):*
<details><summary><i>Previously, omni-viral (Zakarum) wrote…</i></summary><blockquote>

You need to do it yourself
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/amethyst/amethyst/537)*

**Pull request merged**